### PR TITLE
Fix various build breakages

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Hey hi, fix few build breaks introduced in #42230. Seems like none of these were included in the CI tests, I happen to have a board with a st7735r so I actually tested that one. Found another few issues, fixed all I found and added some corresponding test config so hopefully the CI should catch these in the future.

### CODE REVIEW

**Review by erwango — DISMISSED** [review]

File: samples/drivers/can/sample.yaml [hunk 1]
```diff
@@ -9,3 +9,6 @@ tests:
       type: one_line
       regex:
         - "Counter received: (.*)"
+  sample.drivers.can.dfrobot_can_bus_v2_0.ssd1306_128x64:
```
Comment (jfischer-no) [inline samples/drivers/can/sample.yaml:12]: `.ssd1306_128x64` does not seem to be quite right.

Comment (fabiobaltieri) [reply samples/drivers/can/sample.yaml:12]: Ah, sorry, copypaste leftover, fixed.

**Comment by fabiobaltieri** [comment]
Fixed an mangled up test name in can.

**Review by gmarull — APPROVED** [review]
Thanks for the fixes and taking care of increasing CI coverage

**Comment by fabiobaltieri** [comment]
> Thanks for the fixes and taking care of increasing CI coverage

Hey no worries, learned few things about the testing infrastructure... unfortunately the ones in ` tests/drivers/build_all/` don't work quite right, few of these have a stub dts entry for testing under `sensor` but are not enabled in the build. That one may use a bit of a more substantial revisiting/refactoring/expansion... I may look into it if I have some time in the next few days but for this PR extending few framework specific test seemed alright.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**